From fd34de9e4a8b91202f0afb9aa222c06bc9761283 Mon Sep 17 00:00:00 2001
From: Tommaso Tocci <tommaso@tocci.pro>
Date: Mon, 2 Jul 2018 11:54:10 +0200
Subject: [PATCH] sm: clear client disconnection event on socket

---
 src/na/na_sm.c | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/src/na/na_sm.c b/src/na/na_sm.c
index d96f1b6..01e4110 100644
--- a/src/na/na_sm.c
+++ b/src/na/na_sm.c
@@ -2134,6 +2134,16 @@ na_sm_progress_sock(na_class_t *na_class, struct na_sm_addr *poll_addr,
             *progressed = NA_TRUE;
         }
         break;
+        case NA_SM_SOCK_DONE: {
+            *progressed = NA_FALSE;
+            ret = na_sm_poll_deregister(na_class, NA_SM_SOCK, poll_addr);
+            if (ret != NA_SUCCESS) {
+                NA_LOG_ERROR("Could not deregister socket from poll set");
+                ret = NA_PROTOCOL_ERROR;
+                goto done;
+            }
+        }
+        break;
         default:
             /* TODO Silently ignore, no progress */
             *progressed = NA_FALSE;
@@ -2860,7 +2870,7 @@ na_sm_addr_free(na_class_t *na_class, na_addr_t addr)
         ret = na_sm_poll_deregister(na_class, NA_SM_SOCK, na_sm_addr);
         if (ret != NA_SUCCESS) {
             NA_LOG_ERROR("Could not delete sock from poll set");
-            goto done;
+            //goto done;
         }
 
         /* Remove addr from poll addr queue */
-- 
2.18.0

